### ISSUE TITLE
Support hot release upgrades

### ISSUE THREAD

**Issue by weiss** [processone/eturnal#29]
Supporting hot [release upgrades][1] would be nice, as restarting the TURN service kills running calls.

Caveats:

* To avoid an [emulator restart during the upgrade][2], the new eturnal version would have to stick to the same OTP version.  At least the core applications cannot be soft-upgraded: ERTS, Kernel, STDLIB, and SASL.
* Supporting hot upgrades with distribution packages would probably be (too) combersume, as both the old and new version would have to be installed in parallel during the upgrade.  Just using versioned package names won't do the trick, as both the old and the new package would install identical file paths.  (Ab)using mechanisms such as Debian's [diversions][3] or [alternatives][4] is probably too much a hassle given the number of affected files.  However, hot upgrades should be possible with the [binary release][5].

[1]: https://erlang.org/doc/design_principles/release_handling.html
[2]: https://erlang.org/doc/system_principles/upgrade.html
[3]: https://debian.org/doc/debian-policy/ap-pkg-diversions.html
[4]: https://debian.org/doc/manuals/debian-reference/ch02.en.html#_the_update_alternatives_command
[5]: https://eturnal.net/documentation/#Installation
